src/main/java/net/pms/PMS.java

```diff
@@ -788,6 +791,12 @@ public void run() {
 							"-script", database.getDatabasePath()  + File.separator + "logging_report.txt");
 					} catch (SQLException e) {}
 				}
+
+				try (Statement stmt = database.getConnection().createStatement()) {
```

Comment:
Shouldn't we check whether the database is enabled before doing this?

Reply:
@SubJunk What do you mean with "enabled"? The database doesn't shut down on an HUP any more, because I've added `";DB_CLOSE_ON_EXIT=FALSE";` to the database connect URL. Therefore, the database is still online while entering the UMS shutdown hook, where we actually manually close it by calling `SHUTDOWN COMPACT`.

Reply:
I just mean the user might have the database/cache disabled, so we should check `configuration.getUseCache()`

Reply:
I see. Tracing back, which methods could initiate a database connection, by calling `PMS.getDatabase()` I think it's more save to track the initialization process.